src/azure/storage/config.rs

```diff
@@ -16,6 +14,24 @@ pub struct Config {
     ///
     /// - this field if it's `is_some`
     pub sas_token: Option<String>,
-    /// Will load credential from IMDS if it's `is_some`.
-    pub imds_credential: Option<ImdsCredential>,
+    /// Specifies the object id associated with a user assigned managed service identity resource
+    ///
+    /// The values of client_id and msi_res_id are discarded
+    pub imds_object_id: Option<String>,
```

Comment:
Please don't add `imds_` prefix (alought I don't know what's object_id, are you sure it's related to azure storage?).

Reply:
We’ll update soon! The object_id is not related to Azure Storage, but rather to Virtual Machines. We will use it to get an access token. For more details, please refer to this [document](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http).

And just wanted to remind you that this pull request is for authorizing with Azure Active Directory on an Azure Virtual Machine.